### ISSUE TITLE
woocommerce_wp_text_input alignment

### ISSUE THREAD

**Issue by derweili** [woocommerce/woocommerce#22395]
**Describe the bug**
When adding a custom Form field to a product variation via 'woocommerce_wp_text_input' the input fields have wrong alignment if the 'type' option is other than the default ones. e.g. when using `'type' => 'date',`

```
woocommerce_wp_text_input(
      array(
      'id' => 'start_date[' . $loop . ']',
      'class' => 'short',
      'label' => __( 'Start Date: ', 'woocommerce' ),
      'value' => get_post_meta( $variation->ID, 'start_date', true ),
      'type' => 'date',
      'wrapper_class' => ' form-row form-row-first'
      )
);
```

This is error happens because the css used for the text input field has following selector:

`input[type=checkbox], input[type=color], input[type=date], input[type=datetime-local], input[type=datetime], input[type=email], input[type=month], input[type=number], input[type=password], input[type=radio], input[type=search], input[type=tel], input[type=text], input[type=time], input[type=url], input[type=week], select, textarea`

 instead of just `input`

This should be changed. If there is a way to change the input type, there should also be a styling for other input types as well.

**To Reproduce**
1. Hook into `woocommerce_variation_options_pricing`
2. Create a input field via `woocommerce_wp_text_input`
3. Give it the options `'type' => 'date'`

**Screenshots**
It's looking great with the default input type
![bildschirmfoto 2019-01-09 um 13 23 44](https://user-images.githubusercontent.com/6911678/50899260-d9a54f80-1411-11e9-92e6-c939ae43c1b5.png)

It's looking broken wehen using the input type "date"
![bildschirmfoto 2019-01-09 um 13 23 30](https://user-images.githubusercontent.com/6911678/50899289-f17cd380-1411-11e9-95c2-d75ff1e04847.png)

**Expected behavior**
The input field should be aligned the in the same way as all the other input fields.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>

### WordPress Environment ###

Home URL: http://woocommerce-product-type.test
Site URL: http://woocommerce-product-type.test
WC Version: 3.5.3
Log Directory Writable: ✔
WP Version: 5.0.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.37 (Unix) OpenSSL/1.0.2q PHP/7.1.25
PHP Version: 7.1.25 - We recommend using PHP version 7.2 or above for greater performance and security. How to update your PHP version
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.63.0
OpenSSL/1.0.2q

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.11-MariaDB
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.5.3
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 2.90MB
Database Data Size: 2.06MB
Database Index Size: 0.84MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB
wp_comments: Data: 0.02MB + Index: 0.08MB
wp_links: Data: 0.02MB + Index: 0.02MB
wp_options: Data: 1.52MB + Index: 0.02MB
wp_postmeta: Data: 0.02MB + Index: 0.03MB
wp_posts: Data: 0.02MB + Index: 0.06MB
wp_termmeta: Data: 0.02MB + Index: 0.03MB
wp_terms: Data: 0.02MB + Index: 0.03MB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB
wp_usermeta: Data: 0.02MB + Index: 0.03MB
wp_users: Data: 0.02MB + Index: 0.05MB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB

### Post Type Counts ###

page: 6
post: 2
product: 2
product_variation: 1

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (4) ###

Optimus: by KeyCDN – 1.6.0
Wahd Custom Product Type: by YOUR NAME HERE – 0.1.0
WooCommerce: by Automattic – 3.5.3
WP Mail SMTP: by WPForms – 1.4.1

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
training (training)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Shop base: #5 - /index.php/shop/
Cart: #6 - /index.php/cart/
Checkout: #7 - /index.php/checkout/
My account: #8 - /index.php/my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.4.2
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

</details>

**Comment by peterfabian** [comment]
After a bit of digging, I found that the style you mentioned actually comes from WP core, where it was specified as such, because of CSS specificity (so that it would not get overwritten by some other rule). More background for this can be found on the core ticket: https://core.trac.wordpress.org/ticket/28262

However, WC styles these a bit in admin.css, too, thus my proposal for a fix in PR #22590.

Please check if it solves the problem for you.

**Comment by janholpuch** [comment]
Try to use: 
type' => 'date',
'class => 'short date-picker'

It will activate the datapicker over the input.
